File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/LeonardoClient.scala
package org.broadinstitute.dsde.rawls.workspace

import org.broadinstitute.dsde.workbench.client.leonardo.api.AppsV2Api;
import org.broadinstitute.dsde.workbench.client.leonardo.model.CreateAppRequest;
import org.broadinstitute.dsde.workbench.client.leonardo.model.AppType;

import scala.concurrent.Future
import org.broadinstitute.dsde.workbench.client.leonardo.ApiClient

class LeonardoClient(val leonardoBasePath: String) {

  def getAppsV2leonardoApi(accessToken: String): AppsV2Api = {
    val apiClient = new ApiClient()
    apiClient.setAccessToken(accessToken)
    apiClient.setBasePath(leonardoBasePath)
    new AppsV2Api(apiClient)
  }

  def createWDSInstance(token: String, workspaceId: String, appName: String): Unit = {
    val createAppRequest = new CreateAppRequest();
    createAppRequest.setAppType(AppType.CROMWELL);
    getAppsV2leonardoApi(token).createAppV2(workspaceId, appName, createAppRequest);
  }

}

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/LeonardoClient.java
import org.broadinstitute.dsde.workbench.client.leonardo.api.AppsV2Api
import org.broadinstitute.dsde.workbench.client.leonardo.ApiClient
import org.broadinstitute.dsde.workbench.client.leonardo.model.GetAppResponse
import org.broadinstitute.dsde.workbench.client.leonardo.model.CreateAppRequest
import org.broadinstitute.dsde.workbench.client.leonardo.model.AppType

class LeonardoClient(leonardoBasePath: String) {
  def AppsV2leonardoApi(accessToken: String): RuntimesApi = {
    val apiClient = new ApiClient()
    apiClient.setAccessToken(accessToken)
    apiClient.setBasePath(leonardoBasePath)
    new AppsV2Api(apiClient)
  }

  def createWDSInstance(workspaceId: String, appName: String): T = {
    CreateAppRequest createAppRequest = new CreateAppRequest()
    // WDS supports CROMWELL appTypes for now
    createAppRequest.setAppType(AppType.CROMWELL)
    AppsV2leonardoApi appsV2leonardoApi = new AppsV2leonardoApi(token)
    appsV2leonardoApi.createAppV2(workspaceId, appName, createAppRequest)
  }
}